src/qibo/ui/mpldrawer.py

```diff
@@ -0,0 +1,687 @@
+    return path.join(dir_path, filename)  # absolute file path of given file
+
+
+def _check_list_str(substrings, string):
+    return any(item in string for item in substrings)
+
+
+def _process_gates(array_gates):
+    gates_plot = []
+
+    for gate in array_gates:
+        init_label = gate.name.upper()
+
+        if init_label == "CCX":
+            init_label = "TOFFOLI"
+        elif init_label == "CX":
+            init_label = "CNOT"
+        elif _check_list_str(["SX", "CSX"], init_label):
+            is_dagger = init_label[-2:] == "DG"
+            init_label = (
+                r"$\rm{\sqrt{X}}^{\dagger}$" if is_dagger else r"$\rm{\sqrt{X}}$"
+            )
+        elif (
+            len(gate._control_qubits) > 0
+            and "C" in init_label[0]
+            and "CNOT" != init_label
+        ):
+            init_label = gate.draw_label.upper()
+
+        if init_label in [
+            "ID",
+            "MEASURE",
+            "KRAUSCHANNEL",
+            "UNITARYCHANNEL",
+            "DEPOLARIZINGCHANNEL",
+            "READOUTERRORCHANNEL",
+        ]:
+            for qbit in gate._target_qubits:
+                item = (init_label,)
+                item += ("q_" + str(qbit),)
+                gates_plot.append(item)
+        elif init_label == "ENTANGLEMENTENTROPY":
+            for qbit in list(range(circuit.nqubits)):
+                item = (init_label,)
+                item += ("q_" + str(qbit),)
+                gates_plot.append(item)
+        else:
+            item = ()
+            item += (init_label,)
+
+            for qbit in gate._target_qubits:
+                if qbit is tuple:
+                    item += ("q_" + str(qbit[0]),)
+                else:
+                    item += ("q_" + str(qbit),)
+
+            for qbit in gate._control_qubits:
+                if qbit is tuple:
+                    item += ("q_" + str(qbit[0]),)
+                else:
+                    item += ("q_" + str(qbit),)
+
+            gates_plot.append(item)
+
+    return gates_plot
+
+
+def plot(circuit, scale=0.6, cluster_gates=True, style=None):
+    """Main matplotlib plot function for Qibo circuit
+    circuit         A Qibo circuit to plot (type: qibo.models.circuit.Circuit)
+    scale           Scale the ouput plot
+    cluster_gates   Group single-qubit gates
+    style           Style applied to the circuit (built-in styles: garnacha, fardelejo, quantumspain, color-blind and cachirulo)
+
+    ax              An Axes object encapsulates all the elements of an individual plot in a figure (type: matplotlib.axes._axes.Axes)
+    ax.figure       A Figure object (type: matplotlib.figure.Figure)
+    """
+
+    if style is not None:
+        if type(style) is dict:
+            plot_params.update(style)
+        else:
+            plot_params.update(
+                STYLE[style] if style in list(STYLE.keys()) else STYLE["default"]
+            )
+    else:
+        plot_params.update(STYLE["default"])
```

Comment:
```suggestion
	params = PLOT_PARAMS.copy()
    if not isinstance(style, dict):
        try:
            style = STYLE.get(style)
        except AttributeError:
            style = STYLE["default"]
    params.update(style)
```

It could even a function on its own `_plot_params(style: Union[dict, str, None]) -> dict`.